spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/VectorizedSparkParquetReaders.java

```diff
@@ -49,4 +54,45 @@ public static ColumnarBatchReader buildReader(
                 expectedSchema, fileSchema, setArrowValidityVector,
                 idToConstant, ColumnarBatchReader::new));
   }
+
+  public static ColumnarBatchReader buildReader(
+          Schema expectedSchema,
+          MessageType fileSchema,
+          boolean setArrowValidityVector,
+          Map<Integer, ?> idToConstant,
+          DeleteFilter<InternalRow> deleteFilter) {
+    return (ColumnarBatchReader)
+        TypeWithSchemaVisitor.visit(expectedSchema.asStruct(), fileSchema,
+            new ReaderBuilder(
+                expectedSchema, fileSchema, setArrowValidityVector,
+                idToConstant, ColumnarBatchReader::new, deleteFilter));
+  }
+
+  private static class ReaderBuilder extends VectorizedReaderBuilder {
+    private final DeleteFilter<InternalRow> deleteFilter;
+
+    ReaderBuilder(
+        Schema expectedSchema,
+        MessageType parquetSchema,
+        boolean setArrowValidityVector,
+        Map<Integer, ?> idToConstant,
+        Function<List<VectorizedReader<?>>, VectorizedReader<?>> readerFactory,
+        DeleteFilter<InternalRow> deleteFilter) {
+      super(expectedSchema, parquetSchema, setArrowValidityVector, idToConstant, readerFactory);
+      this.deleteFilter = deleteFilter;
+    }
+
+    @Override
+    protected VectorizedReader<?> vectorizedReader(List<VectorizedReader<?>> reorderedFields) {
```

Comment:
We can probably simplify a little bit like [here](https://github.com/apache/iceberg/pull/3287/files#r743032246).